source/util/pooled_linked_list.h

```diff
@@ -91,39 +183,45 @@ class PooledLinkedList {
     return false;
   }
 
-  // Moves the elements in the provided list into this pool.
-  // Provides a way to compact the pool, reclaiming unused storage.
-  void move_to(Head& list_head, PooledLinkedList& that) {
+  // Returns the PooledLinkedListNodes that owns this list's nodes.
+  NodePool& pool() { return nodes_; }
+
+  // Moves the nodes in this list into |new_pool|, providing a way to compact
+  // storage and reclaim unused space.
+  //
+  // Upon completing a sequence of |move_nodes| calls, you must swap storage
+  // from |new_pool| into the pool used by your PooledLinkedLists.
+  // Example usage:
+  //
+  //    NodePool old_pool;  // Existing lists use this pool
+  //    NodePool new_pool;  // Temporary storage
+  //    for (PooledLinkedList& list : lists) {
+  //        list.move_to(new_pool);
+  //    }
+  //    old_pool = std::move(new_pool);
```

Comment:
This seems like a big pitfall.  People could easily forget  to do this.  Or they not move all of  the lists to the new pool before calling std::move.  The problem is that it is not always true that the nodes in the list refer to nodes in the pool `nodes_`.  We should try to always make that true.

I wonder if it might be better to make `nodes_` in this class a pointer.  Then you can make `used_id_pool_` and `use_pool_` unique pointers.  Then from the perspective of the caller, they do not have to do anything to make sure that nodes that the list refer to are in `nodes_`.  That will always be true.